lib/src/execution/windows/execution.cc

```diff
@@ -358,6 +358,19 @@ namespace facter { namespace execution {
         option_set<execution_options> const& options,
         uint32_t timeout)
     {
+        // Since we use a job object in the windows world, we want to
+        // be sure we're not in a job object, or at least able to
+        // break our processes out if we are in one.
+        BOOL in_job = FALSE;
+        IsProcessInJob(GetCurrentProcess(), nullptr, &in_job);
+        if (in_job) {
+            JOBOBJECT_BASIC_LIMIT_INFORMATION limits;
+            QueryInformationJobObject(nullptr, JobObjectBasicLimitInformation, &limits, sizeof(limits), nullptr);
+            if (!(limits.LimitFlags & JOB_OBJECT_LIMIT_BREAKAWAY_OK)) {
```

Comment:
Well, timeouts will still work, just that the child process can't be terminated (something we could certainly log a warning for if it occurs; e.g. "cannot terminate process <ID> because a break-away job could not be created" or some such).

This effectively improves the error message but still prevents facter from running when part of a restrictive job.  Would that mean the transient CI failure we were seeing as a result of this problem still occur or are we expecting tests/acceptance to start ignoring when it fails with this error message?